src/ci/docker/host-x86_64/mingw-check/Dockerfile

```diff
@@ -40,5 +40,5 @@ ENV SCRIPT python3 ../x.py --stage 2 test src/tools/expand-yaml-anchors && \
            /scripts/validate-toolstate.sh && \
            /scripts/validate-error-codes.sh && \
            # Runs checks to ensure that there are no ES5 issues in our JS code.
-           es-check es5 ../src/librustdoc/html/static/js/*.js && \
+           es-check es6 ../src/librustdoc/html/static/js/*.js && \
```

Comment:
We wanted to make this change in bigger PR but I guess it's fine... Don't forget to update the comment just above please.

Reply:
I actually think a giant PR adding ES6 features wherever possible, and updating the linter at the same time would be a bad idea. The way we should do it is like I'm doing here:

 - update the linter to accept es6 as a standalone commit
 - start using es6 features where they are useful
 - do a series of standalone commits that do one of two things:
   - adopt an ES6 feature where it is useful (e.g. const everything that can be consted), or
   - migrate a discrete chunk of code to using various useful ES6 features.

Reply:
Like I said, it's fine. :wink: